contents/metropolis/code/python/metropolis.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+
+
+def f(x):
+    '''Function proportional to target distribution, a sum of Gaussians'''
+    # Gaussian heights, width parameters, and mean positions respectively:
+    a1, b1, x1 = 10,   4, -4
+    a2, b2, x2 =  3, 0.2, -1
+    a3, b3, x3 =  1,   2,  5
+    
+    return a1*np.exp( -b1*(x-x1)**2 ) + \
+           a2*np.exp( -b2*(x-x2)**2 ) + \
+           a3*np.exp( -b3*(x-x3)**2 )
+
+def g():
+    '''Random step vector'''
+    return np.random.uniform(-1,1)
+
+def iterate(x, f=f, g=g):
+    '''Perform one full iteration and return new position'''
+    
+    x_proposed = x + g()
+    A = min(1, f(x_proposed)/f(x) )
```

Comment:
PEP8 recommends snake_case for variables in Python
```suggestion
    a = min(1, f(x_proposed) / f(x))
```